nvflare/app_common/workflows/scatter_and_gather_autofedrl.py

```diff
@@ -0,0 +1,477 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import traceback
+
+from nvflare.apis.client import Client
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_constant import ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.impl.controller import ClientTask, Controller, Task
+from nvflare.apis.shareable import Shareable
+from nvflare.apis.signal import Signal
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.abstract.learnable_persistor import LearnablePersistor
+from nvflare.app_common.abstract.shareable_generator import ShareableGenerator
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.app_event_type import AppEventType
+from nvflare.fuel.utils import fobs
+from nvflare.fuel.utils.import_utils import optional_import
+from nvflare.widgets.info_collector import GroupInfoCollector, InfoCollector
+
+
+class ScatterAndGatherAutoFedRL(Controller):
```

Comment:
I see a lot of duplicate codes VS ScatterAndGather, maybe we can inherit from ScatterAndGather and overwrites the methods that need to be changed.